### PR TITLE
DOCSP-37502 - Upgrade

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-37502
Staging - https://preview-mongodbmongokart.gatsbyjs.io/pymongo/docsp-37502-upgrade/upgrade/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST? (Link error commented out in page)
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/upgrade.txt [hunk 1]
```diff
@@ -23,41 +20,51 @@ Upgrade Driver Versions
...
-
-.. code-block:: python
+This page describes the changes you must make to your application
+when you upgrade to a new version of {+driver-short+}.
 
-   # "insert.py" (with PyMongo 3.X)
-   from pymongo import MongoClient
+.. important:: 
+  
+   This guide includes breaking changes only to {+driver-short+} versions after v4.0.
```
Comment (jordan-smith721) [inline source/upgrade.txt:28]:
```suggestion
   This guide includes breaking changes to only {+driver-short+} versions after v4.0.
```

File: source/upgrade.txt [hunk 2]
```diff
@@ -23,41 +20,51 @@ Upgrade Driver Versions
...
+   If you're upgrading from {+driver-short+} v2 or v3, see the
+   `{+driver-short+} 4 Migration Guide. <https://pymongo.readthedocs.io/en/latest/migrate-to-pymongo4.html>`__
+
+Before you upgrade, perform the following actions:
+
+- Ensure the new {+driver-short+} version is compatible with the MongoDB Server versions
```
Comment (jordan-smith721) [inline source/upgrade.txt:34]:
```suggestion
- Ensure the new {+driver-short+} version is compatible with the MongoDB Server version
```

File: source/upgrade.txt [hunk 3]
```diff
@@ -23,41 +20,51 @@ Upgrade Driver Versions
...
+  your application connects to and the Python version your
+  application runs on. See the
+  :ref:`{+driver-short+} Compatibility <pymongo-compatibility>`
+  page for this information.
```
Comment (jordan-smith721) [inline source/upgrade.txt:38]:
```suggestion
  application runs on. For version compatibility information, see the
  :ref:`{+driver-short+} Compatibility <pymongo-compatibility>`
  page.
```

File: source/upgrade.txt [hunk 4]
```diff
@@ -23,41 +20,51 @@ Upgrade Driver Versions
...
+- Address any breaking changes between the current version of the driver
+  your application is using and your planned upgrade version in the
+  :ref:`Breaking Changes <pymongo-breaking-changes>` section.
```
Comment (jordan-smith721) [inline source/upgrade.txt:41]:
```suggestion
- Address any breaking changes between the driver version
  your application is using and your planned upgrade version in the
  :ref:`Breaking Changes <pymongo-breaking-changes>` section.
```

File: source/upgrade.txt [hunk 5]
```diff
@@ -137,939 +150,17 @@ as shown in the following example:
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-``~bson.int64.Int64``, ``~bson.min_key.MinKey``,
-``~bson.max_key.MaxKey``, ``~bson.timestamp.Timestamp``,
-``~bson.regex.Regex``, and ``~bson.dbref.DBRef`` now implement
-``__slots__`` to reduce memory usage. This means that their attributes are fixed, and new
-attributes cannot be added to the object at runtime.
\ No newline at end of file
+- To improve support for the `Pyright <https://github.com/microsoft/pyright>`__ tool,
+  the ``ClientSession`` class no longer uses generic typing,
```
Comment (jordan-smith721) [inline source/upgrade.txt:157]:
```suggestion
  the ``ClientSession`` class no longer uses generic typing.
```

**Review by jordan-smith721 — APPROVED** [review]
LGTM with a couple suggestions

File: source/upgrade.txt (hunk 2 above)
Comment (mongoKart) [reply source/upgrade.txt:34]: I thought about this when writing it, but I think you could have a single application (or multiple instances of the same application) connecting to multiple deployments

**Review by blink1073 — APPROVED** [review]
LGTM!

**Pull request merged**